src/module/item/abstract-effect/document.ts

```diff
@@ -12,6 +12,9 @@ export abstract class AbstractEffectPF2e extends ItemPF2e {
     abstract increase(): Promise<void>;
     abstract decrease(): Promise<void>;
 
+    /** If true, the AbstractEffect should be hidden from the user unless they are a GM */
+    abstract get secret(): boolean;
```

Comment:
Instead of making this abstract and overriding to be return false in the condition, maybe make it return false here, and then leave it to subclasses to decide if they wanna allow secret versions or not.

Reply:
Aye, changed to that.